### PR TITLE
Hashicorp vault integration

### PR DESCRIPTION
**Pull request by liorblob**
## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/14017

## Must have
- [X] Tests
- [X] Documentation - https://github.com/demisto/etc/issues/14718
- [X] Code Review

### CODE REVIEW

File: Integrations/integration-HashiCorp-Vault.yml [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+        if not isinstance(ENGINES, list):
+            ENGINES = [ENGINES]
+
+        if len(ENGINES) == 0:
+            return_error('No secrets engines specified')
+
+        for engine_type in ENGINES:
+            engines_to_fetch = list(filter(lambda e: e['type'] == engine_type, ENGINE_CONFIGS))
+            if len(engines_to_fetch) == 0:
+                return_error('Engine type not configured')
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: Please add to the error message that need to be configured in a command

File: Integrations/integration-HashiCorp-Vault.yml [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
+    description: Enable a new secrets engine at the given path.
+  - name: hashicorp-configure-engine
+    arguments:
+    - name: path
+      required: true
+      description: The engine path, e.g. secret/
+    - name: type
+      required: true
+      description: The engine type, e.g. KV
+    - name: version
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: Shouldn't this be required or have an initial value?

File: Integrations/integration-HashiCorp-Vault.yml [hunk 3]
```diff
@@ -0,0 +1,1042 @@
...
+      auto: PREDEFINED
+      predefined:
+      - "1"
+      - "2"
+      description: The engine version (for KV engines)
+    description: Configure a secrets engine to fetch secrets from.
+  - name: hashicorp-reset-configuration
+    arguments: []
+    description: Reset the engines configuration
+  dockerimage: blob
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: Don't forget to change docker image

File: Integrations/integration-HashiCorp-Vault.yml [hunk 4]
```diff
@@ -0,0 +1,1042 @@
...
...56vDC28jmaLnbkQZ9VSoJhqxzzh08rmhguyXPqI9xxINzOotPsDkhbXavMZlWmZgeVjh93gbHYx5//kWYFZQkfLg0P2hAg3+M3xwQb1GFe0nacVaDz3XOB3m9xiU5UP64i/M9rzG5dq+6xnanqsZ2u4rc4OEX+P3eBt9ZYQRgnf/4IAwERe7IsyARnl1PBTX9IRKYteD7mNwfNb5gD+4nKG9/9fcIOHX+D3eRggQbp42Pp616wYBUVeaVtwPhM8+H/iuPzODhF/j94QB4RNaFviR1h1K6wPxYkcyZvqHUQckC2y6utN3bzWatzDKqtZGHZDcuh56oWNSH4h3dqRRSdP5frsfXUAkVn1vNTxyO+V5zfG6GGe/ohMIn1BOKFJEej5NBfKEwfHRAWE0OBL6ldh+0H4ussRpOqVXrfFQRdtJneqN+7qD+G9J8N7hbkzEalRBusynrDFuiE7lzbCbK990s6s/Ns732WGkweMMQgxs95JSB0BnzPF61CHKWVC+cY1rZWAG5fmuhs1Vlq96JprsbzLyZRmoL1gaADrZGxHHsiSwKagwwUtDLmV9rBEwKiasvb9sY0FezgIRx/BsUybCoN5Bz32G2CppZLlJTfNssrO94sEoG6MrCLIytN8KS/EB/CQBNS1WCflNIJgeOHvKfcrLlULd2YT5ibwZ5zHxB+JAicbBjgg1MV5KcjcuIk/TdFBrKnwzTsCcyk6LmIynBd04f0Lu1qctNi05CZ2dtbxkKiknyNmQaA0Il5qOTkLDmZyXrQDzMyRNLACPmgmlu+Ebt0MTG991Yhs6sUM34n6nvufYTfsMJ4G9VMjGigNR0+LRbTf84P4lW31wWdOtZlCx8qUpMPw9fZEyUxyI1rYpCA8uGQ5Kkr8I+5y39RFa4Ds8QiCXwVpZYk2q1C9FeIexw/BZm5yMmY01PbgBJ7egcF3oLxnWjl1Kc4Ty6USAPRBDSbTYGpZqacMGJf8TvnqBxEaS8yLqyKpJhnyqfzRgHPrzzzQsfIWweE0CuHw12euHMpVG5f7N+Nwiuh2Yf77hbj6mf84Rap6/YoO1x2cnL3m1Dp2e0h1RwtkuB5CSxhZGPL9zKj4UONX/NncAxEWS/E/FOcaSJ+mfHDlNhNFlrTP0l5x9weL38LF0GrWHq2m2/uLGpT4pWvX/zRDqQjZWrVoAAAAASUVORK5CYII=
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: Don't forget to change logo

File: Integrations/integration-HashiCorp-Vault.yml (hunk 2 above)
Comment (liorblob) [reply Integrations/integration-HashiCorp-Vault.yml]: No, because it only applies to KV engine

**Pull request merged**